### DISCUSSION TITLE
How to use matchesRegexp with JSON response

### DISCUSSION THREAD

**Discussion by WillGibson** [artilleryio/artillery#1771] [Artillery]
Let's say I have a response like this in the debug output...

```
  Body:
    "{\"authId\":\"FZ5VXpVeFVFSk1RVzh4UlRsbk1sQlVValpRZUV.KWVdYQnll\",\"someOtherProperty\":[{\"some value\"}"
```

...and I want to check that the authId contains alphanumeric characters and dots.

I this this expectation should work...

```
- matchesRegexp: '("authId":"[0-9-A-z.]*")'
```

...but it doesn't.

What am I doing wrong?

**Comment by hassy** [comment]
hi @WillGibson 👋 that looks like a regexp that *should* be working - how are you checking that it doesn't?

**Reply by WillGibson** [reply]
When I run the scenario the output includes these...

```
  plugin:expect checker: matchesRegexp +2ms
  plugin:expect check valid regexp +0ms
* POST /blah/blah
  ok statusCode 200 
  ok hasProperty authId 
  ok equals capture_email, capture_email 
  not ok matchesRegexp false 
  expected: true
       got: false
```

**Reply by hassy** [reply]
Thanks @WillGibson. I'll need to test it in the plugin rather than just in the Node.js REPL. Is the body always guaranteed to be a single line? (i.e. with no newline characters anywhere)

**Reply by WillGibson** [reply]
Yes, I think so. It's just a JSON string, which is output in the Artillery output in double quotes with all it's double quotes escaped.
